ddtrace/contrib/anthropic/_streaming.py

```diff
@@ -0,0 +1,320 @@
+        )
+        return traced_stream
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
+
+
+def _process_finished_stream(integration, span, args, kwargs, streamed_chunks):
+    # builds the response message given streamed chunks and sets according span tags
+    resp_message = {}
+    try:
+        resp_message = _construct_message(streamed_chunks)
+
+        if integration.is_pc_sampled_span(span):
+            _tag_streamed_chat_completion_response(integration, span, resp_message)
+        if integration.is_pc_sampled_llmobs(span):
+            integration.llmobs_set_tags(
+                span=span,
+                resp=resp_message,
+                args=args,
+                kwargs=kwargs,
+            )
+    except Exception:
+        log.warning("Error processing streamed completion/chat response.", exc_info=True)
+
+
+def _construct_message(streamed_chunks):
+    """Iteratively build up a response message from streamed chunks.
+
+    The resulting message dictionary is of form:
+      {"content": [{"type": [TYPE], "text": "[TEXT]"}], "role": "...", "finish_reason": "...", "usage": ...}
+    """
+    message = {"content": []}
+    for chunk in streamed_chunks:
+        message = _extract_from_chunk(chunk, message)
+
+        if "finish_reason" in message:
+            return message
+    return message
+
+
+def _extract_from_chunk(chunk, message) -> Tuple[Dict[str, str], bool]:
+    """Constructs a chat message dictionary from streamed chunks given chunk type"""
+    TRANSFORMATIONS_BY_BLOCK_TYPE = {
+        "message_start": _on_message_start_chunk,
+        "content_block_start": _on_content_block_start_chunk,
+        "content_block_delta": _on_content_block_delta_chunk,
+        "message_delta": _on_message_delta_chunk,
+        "error": _on_error_chunk,
+    }
+    chunk_type = getattr(chunk, "type", "")
+    transformation = TRANSFORMATIONS_BY_BLOCK_TYPE.get(chunk_type)
+    if transformation is not None:
+        message = transformation(chunk, message)
+
+    return message
+
+
+def _on_message_start_chunk(chunk, message):
+    # this is the starting chunk of the message
+    if getattr(chunk, "type", "") != "message_start":
+        return message
+
+    chunk_message = getattr(chunk, "message", "")
+    if chunk_message:
+        chunk_role = getattr(chunk_message, "role", "")
+        chunk_usage = getattr(chunk_message, "usage", "")
+        if chunk_role:
+            message["role"] = chunk_role
+        if chunk_usage:
+            message["usage"] = {}
+            message["usage"]["input_tokens"] = getattr(chunk_usage, "input_tokens", 0)
+            message["usage"]["output_tokens"] = 0
+    return message
+
+
+def _on_content_block_start_chunk(chunk, message):
+    # this is the start to a message.content block (possibly 1 of several content blocks)
+    if getattr(chunk, "type", "") != "content_block_start":
+        return message
```

Comment:
Are these checks necessary if we already do the check in `_extract_from_chunk()`?